src/core/providers/ogr/qgsogrprovider.cpp

```diff
@@ -1445,7 +1445,8 @@ size_t QgsOgrProvider::layerCount() const
 QgsWkbTypes::Type QgsOgrProvider::wkbType() const
 {
   QgsWkbTypes::Type wkb = QgsOgrUtils::ogrGeometryTypeToQgsWkbType( mOGRGeomType );
-  if ( mGDALDriverName == QLatin1String( "ESRI Shapefile" ) && ( wkb == QgsWkbTypes::LineString || wkb == QgsWkbTypes::Polygon ) )
+  const auto wkbFlat = QgsWkbTypes::flatType( wkb );
```

Comment:
```suggestion
  const QgsWkbTypes::Type wkbFlat = QgsWkbTypes::flatType( wkb );
```

(consensus on previous discussions of this nature was that in QGIS code auto should only be used for complex types such as iterators)

Reply:
done